eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala

```diff
@@ -260,6 +266,20 @@ class Router(val nodeParams: NodeParams, watcher: typed.ActorRef[ZmqWatcher.Comm
       stay() using Validation.handleChannelValidationResponse(d, nodeParams, watcher, r)
 
     case Event(WatchExternalChannelSpentTriggered(shortChannelId), d) if d.channels.contains(shortChannelId) || d.prunedChannels.contains(shortChannelId) =>
```

Comment:
Good point! since the scid of a splice won't change until it's confirmed then we can wait 12 blocks after the spend confirms before removing it. 

I like the idea of updating the semantics of `WatchExternalChannelSpentTriggered` to trigger 12 blocks after the spending tx confirms. I think using `WatchConfirmed` might cause interactions with unrelated parts of the code.

Reply:
On the other hand, `WatchConfirmed` already does exactly what we want.. I'll have to investigate if there really would be any interactions.

Reply:
I've now refactored the logic based on our discussion, see comment https://github.com/ACINQ/eclair/pull/2936#issuecomment-2505822454.

I removed `WatchSpentBasic' because now we must wait for an external channel to confirm to know if it was closed or spliced. `WatchExternalChannelSpent` now extends `WatchSpentTriggered`.